isisd/isis_main.c

```diff
@@ -181,6 +181,36 @@ static const struct frr_yang_module_info *const isisd_yang_modules[] = {
 };
 /* clang-format on */
 
+
+static void isis_config_end(void)
+{
+	/* If ISIS config processing thread isn't running, then
+	 * we can return and rely it's properly handled.
+	 */
+	if (!thread_is_scheduled(t_isis_cfg))
+		return;
+
+	THREAD_OFF(t_isis_cfg);
+
+	struct listnode *node, *inode;
+	struct isis *isis;
+	struct isis_area *area;
+
+	for (ALL_LIST_ELEMENTS_RO(im->isis, inode, isis)) {
+		for (ALL_LIST_ELEMENTS_RO(isis->area_list, node, area))
+			config_end_lsp_generate(area);
+	}
+}
+
+static void isis_config_start(void)
+{
+	/* Max wait time for config to load before generating lsp */
+#define ISIS_PRE_CONFIG_MAX_WAIT_SECONDS 5
```

Comment:
Would like feedback on this value

Reply:
I guess the question becomes in my mind, why have a timer?  We have bookends for config start and end and I would prefer that we find and fix all the places that we don't send the config end marker and fix those.  Imagine a scenario where there is a large config, say 100k lines of route-maps and prefix lists, at the start and the isis configuration at the end.  This 5 second timer will likely pop before any isis configuration is read.  Thus doing the wrong thing.  On the other hand this would work for 99.99% of configurations out there without any sort of issue at all, we are unlikely to ever see a bug associated with this at all.

Reply:
I'd like a way to check if a config is currently being read in, which is what I added the timer thread for. I'm open to any alternative approach though

Reply:
ok looking at bgp they have a 600 second timer.  Let's make it the same.

Reply:
Updated